src/ServiceProvider.php

```diff
@@ -73,16 +73,56 @@ public function boot()
         });
     }
 
+    protected function legacyPermissionKey($permission, $resource)
+    {
+        return match ($permission) {
+            'view' => "View {$resource->plural()}",
+            'edit' => "Edit {$resource->plural()}",
+            'create' => "Create new {$resource->singular()}",
+            'delete' => "Delete {$resource->singular()}"
+        };
+    }
+
+    protected function permissionKey($permission, $resource)
+    {
+        if (!$permissionKey = config("runway.permission_keys.{$permission}")) {
+            return $this->legacyPermissionKey($permission, $resource);
+        }
+
+        return str_replace('{resource}', $resource->handle(), $permissionKey);
+    }
+
+    protected function permissionLabel($permission, $resource)
+    {
+        $translationKey = "runway.permissions.{$permission}";
+
+        $label = trans($translationKey, [
+            'resource' => $resource->name()
+        ]);
+
+        if ($label == $translationKey) {
+            return $this->legacyPermissionKey($permission, $resource);
```

Comment:
I know you're trying to cleanup the code a little but can we move the code from the `legacyPermissionKey` method into this method?

Reply:
I mean, it can be done. I'm just confused as why this would be beneficial...

Reply:
Oh... I see... Is it that you don't want a method call to something related to the key in the method that handles the label? In this case I think it makes sense.

Reply:
In fact, leave it how it is. Sorry!